doc/onboarding_guide.md

```diff
@@ -0,0 +1,95 @@
+
+Reach out to `azpycli@microsoft.com` to get the process started. You'll be assigned a dev contact on the CLI team. Early and frequent communication with this contact is essential to ensuring a smooth onboarding.
+
+##Extension vs. Module
+
+One of the key decisions you will need to make is whether to create your commands in a CLI module or an extension.
+
+####Extensions
+
+|                      PROS                      |                         CONS                         |
+|:----------------------------------------------:|:----------------------------------------------------:|
+| Release separate from the CLI release schedule | Requires dedicated installation (az extension add …) |
+| Higher velocity fixes possible                 | Can be broken by changes to the azure-cli-core       |
+| Experimental UX is permissible                 |                                                      |
+
+####CLI Modules
+
+|                                   PROS                                  |                                                                             CONS                                                                             |
+|:-----------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+| Comes automatically with the CLI. No dedicated installation required.   | Strictly tied to CLI release schedule                                                                                                                        |
+| Unlikely to be broken by changes to azure-cli-core (with test coverage) | STRICTLY tied to CLI authoring guidelines. Experimental patterns that may be allowed in extensions could be rejected entirely for inclusion as a CLI module. |
+
+- Common uses for extensions include experimental commands, commands in private or public preview, or to separate between frequently used and rarely used functionality (where infrequently used commands are acquired via extension).
+- Note that if you are trying to get commands into the CLI out of step with the normal release cycle, extensions are your **only** option.
+- Because non-standard, experimental authoring patterns are permitted for extensions, simply trying to "move an extension into the CLI" is often **not** a trivial process and will necessitate a full review with higher scrutiny from the CLI team. Expect to need to make changes.
+
+##Initial Timeline and Milestones
+
+- **Initial Kickoff:** Reach out to your contact on the CLI team. Set up a short 30 minute Teams call to discuss initial questions.
+
+- **Initial Review:** Create a few commands. Schedule a short 30 minute Teams screen-share to demo your commands. This is crucial! Teams have created entire modules using anti-patterns, resulting in comment-filled PRs and last-minute rework! A quick command review would have prevented this.
```

Comment:
Maybe mention that the underlying command methods don't need to be (fully) implemented, just the skeleton (parameter options and group / command structure) is sufficient for the command review?